desc/optimize/optimizer.py

```diff
@@ -231,8 +231,11 @@ def optimize(  # noqa: C901 - FIXME: simplify this
                     "yellow",
                 )
             )
-
-        x0 = objective.x(*things)
+        # we have to use this cumbersome indexing in this method when passing things
+        # to objective to guard against the passed-in things having an ordering
+        # different from objective.things, to ensure the correct order is passed
+        # to the objective
+        x0 = objective.x(*[things[things.index(t)] for t in objective.things])
```

Comment:
So "things" here can be a list of objects from different classes like Equilibrium and FourierRZToroidalSurface. What is meaning of the *before things? I see this a lot in the code.

Reply:
In python `fun(*list)` will "unpack" the list , so that if say we have `list=[a,b,c]`, calling `fun(*list)` is the same thing as calling `fun(a,b,c)`  i.e. it unpacks the iterable and passes the arguments to the function being called

Reply:
and since you will see this elsewhere, the single `*` unpacks a list or tuple, while a double `**` will unpack a dict, i.e `fun(**dict)` where `dict={"a":1, "b":2, "c":3}` will be the same thing as calling `fun(a=1, b=2, c=3)` i.e calling the function using the dict keys as  keyword args and the corresponding dict values as the values passed to the keywords